dev-docs/007-transforms-rfc.md

```diff
@@ -0,0 +1,99 @@
+- Start Date: 2019-09-05
+- RFC PR: 
+
+# Summary
+
+It is desirable to enable arbitrary transform support common in robotic systems. While XVIZ models some explicitly
+with our coordinate types IDENTITY, GEOGRAPHIC, and VEHICLE_RELATIVE, the need for client controlled frames of reference
+will make it easier to interop with other systems as well as match a familiar modeling automous system data.
+
+# Motivation
+
+Some PERSISTENT data, such as *maps* and *point clouds*, could be encoded in various ways such as vehicle relative.
```

Comment:
Possibly expanding on the motivation/background:

It seems to me that the key driver here is a wish to remove absolute coordinates from the stream and send everything including maps and point clouds as vehicle relative (perhaps removing the concept of an "arbitrary" starting pose)?

Of course, maps and pre-generated point clouds will have to be stored with some form of absolute coordinates, so that transformation to vehicle relative is then done during xviz stream creation. I.e the xviz converter must know the relation between vehicle and absolute when emitting the static streams. And it presumably knows the relation between absolute and relative for every frame. 

But we don't want to show the existing of absolute coordinates in the stream....

And we don't even want there to be a fixed origin in the stream?

Reply:
It may be more general to say we want to enable data to be sent once, and separate out how that data is transformed. And specifically avoid resending data when all we need is to send a new transform. 

regarding the *arbitrary* starting pose, one related request is the ability to support multiple vehicles. Right now we have a singular special stream  `/vehicle_pose` and that is it.  But this proposal may be a good way to unlock that functionality as well.

To your point on the absolute coordinates and what pieces know what when.  On the autonomy side I get the sense that there are different interactions with the data, so what is used on vehicle may not be the same as that which is stored elsewhere (like maps) and they store data in some absolute form, but during operation use a local reference for the current task (rather than transform their local data into a different frame).  So at some level it comes down to what hoops we make them jump through vs how they naturally model their data.

The frame of reference really seems to come down to a combination of convenience and performance. We could have said XVIZ requires a single global frame of reference and put the burden on the conversion side, but scope of context at time of conversion may not have everything to do that. Plus the ability to mix different systems together is very useful.

However the more conversations happen the more the flexibility of define arbitrary user-defined frames of references seems desired.

We do not want to model articulated platforms, but provide a familiar mechanism that the autonomy engineers use today.  Modeling the entire scene in a graph with individual objects is an explicit non-goal. . I should probably add that.